test/reduce.jl

```diff
@@ -118,12 +118,12 @@ sum2(itr) = invoke(sum, Tuple{Any}, itr)
 plus(x,y) = x + y
 sum3(A) = reduce(plus, A)
 sum4(itr) = invoke(reduce, Tuple{Function, Any}, plus, itr)
-sum5(A) = reduce(plus, 0, A)
+sum5(A) = reduce(plus, A; int=0)
 sum6(itr) = invoke(reduce, Tuple{Function, Int, Any}, plus, 0, itr)
```

Comment:
Can someone help me with this? What's the form for `invoke` with keyword arguments?

Reply:
I believe you have to use the gensym'd name, i.e. `Symbol("kw##reduce")`, then specify the keyword arguments' types with a `NamedTuple` type. See for example how functions with keyword arguments are precompiled in `base/precompile.jl`.

Reply:
OK, it took me a while to grok but I think I've got this sorted now. :)

Reply:
Does this not work? `invoke(reduce, Tuple{Function, Any}, plus, itr, init=0)`